### ISSUE TITLE
[RuntimeException] Failed to execute git clone --no-checkout

### ISSUE THREAD

**Issue by Asterovim** [drupal-composer/drupal-composer-project#458]
Hi, i have tested with two environment (linux online hosting and a local windows computer).
PHP 7.2 and have 1GB.

```
composer create-project drupal-composer/drupal-project:8.x-dev composered10 --no-interaction
Installing drupal-composer/drupal-project (8.x-dev 4229acea4bb6181f421e8dd0e72a02ccdc98df1d)
  - Installing drupal-composer/drupal-project (8.x-dev 4229ace): Cloning 4229acea4b from cache
    Failed to download drupal-composer/drupal-project from source: Failed to clone https://github.com/drupal-composer/drupal-project.git via https, ssh protocols, aborting.

- https://github.com/drupal-composer/drupal-project.git
  Cloning into 'composered10'...
  done.
  fatal: unable to create thread: Resource temporarily unavailable
  fatal: cannot repack to clean up

- git@github.com:drupal-composer/drupal-project.git
  Cloning into 'composered10'...
  done.
  fatal: unable to create thread: Resource temporarily unavailable
  fatal: cannot repack to clean up

    Now trying to download from dist
  - Installing drupal-composer/drupal-project (8.x-dev 4229ace): Loading from cache
Created project in composered10
> DrupalProject\composer\ScriptHandler::checkComposerVersion
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 131 installs, 0 updates, 0 removals
  - Installing cweagans/composer-patches (1.6.5): Loading from cache
  - Installing composer/installers (v1.6.0): Loading from cache
  - Installing composer/semver (1.4.2): Loading from cache
  - Installing drupal-composer/drupal-scaffold (2.5.4): Loading from cache
  - Installing paragonie/random_compat (v2.0.18): Loading from cache
  - Installing symfony/polyfill-php70 (v1.10.0): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.10.0): Loading from cache
  - Installing symfony/http-foundation (v3.4.22): Loading from cache
  - Installing symfony/polyfill-ctype (v1.10.0): Loading from cache
  - Installing psr/log (1.1.0): Loading from cache
  - Installing symfony/debug (v3.4.22): Loading from cache
  - Installing symfony/console (v3.4.22): Loading from cache
  - Installing ralouphie/getallheaders (2.0.5): Loading from cache
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing guzzlehttp/psr7 (1.5.2): Loading from cache
  - Installing guzzlehttp/promises (v1.3.1): Loading from cache
  - Installing guzzlehttp/guzzle (6.3.3): Loading from cache
  - Installing symfony/finder (v3.4.22): Loading from cache
  - Installing symfony/yaml (v3.4.22): Loading from cache
  - Installing drupal/console-extend-plugin (0.9.2): Loading from cache
  - Installing twig/twig (v1.37.1): Downloading (100%)
  - Installing symfony/translation (v3.4.22): Downloading (100%)
  - Installing symfony/process (v3.4.22): Downloading (100%)
  - Installing symfony/event-dispatcher (v3.4.22): Downloading (100%)
  - Installing psr/container (1.0.0): Downloading (100%)
  - Installing symfony/dependency-injection (v3.4.22): Downloading (100%)
  - Installing doctrine/collections (v1.5.0): Downloading (100%)
  - Installing doctrine/lexer (v1.0.1): Downloading (100%)
  - Installing doctrine/annotations (v1.6.0): Downloading (100%)
  - Installing zendframework/zend-stdlib (3.2.1): Downloading (100%)
  - Installing zendframework/zend-escaper (2.6.0): Downloading (100%)
  - Installing zendframework/zend-feed (2.11.0): Downloading (100%)
  - Installing zendframework/zend-diactoros (1.8.6): Downloading (100%)
  - Installing typo3/phar-stream-wrapper (v2.0.1): Downloading (100%)
  - Installing symfony/validator (v3.4.22): Downloading (100%)
  - Installing symfony/serializer (v3.4.22): Downloading (100%)
  - Installing symfony/routing (v3.4.22): Downloading (100%)
  - Installing symfony/psr-http-message-bridge (v1.1.0): Downloading (100%)
  - Installing symfony/polyfill-iconv (v1.10.0): Downloading (100%)
  - Installing symfony/http-kernel (v3.4.22): Downloading (100%)
  - Installing symfony/class-loader (v3.4.22): Downloading (100%)
  - Installing symfony-cmf/routing (1.4.1): Downloading (100%)
  - Installing stack/builder (v1.0.5): Downloading (100%)
  - Installing masterminds/html5 (2.5.0): Downloading (100%)
  - Installing egulias/email-validator (1.2.15): Downloading (100%)
  - Installing easyrdf/easyrdf (0.9.1): Downloading (100%)
  - Installing doctrine/reflection (v1.0.0): Downloading (100%)
  - Installing doctrine/event-manager (v1.0.0): Downloading (100%)
  - Installing doctrine/cache (v1.8.0): Downloading (100%)
  - Installing doctrine/persistence (v1.1.0): Downloading (100%)
  - Installing doctrine/inflector (v1.3.0): Downloading (100%)
  - Installing doctrine/common (v2.10.0): Downloading (100%)
  - Installing asm89/stack-cors (1.2.0): Downloading (100%)
  - Installing drupal/core (8.6.9): Downloading (100%)
  - Installing zaporylie/composer-drupal-optimizations (1.0.2): Downloading (100%)
  - Installing symfony/dom-crawler (v3.4.22): Downloading (100%)
  - Installing symfony/css-selector (v3.4.22): Downloading (100%)
  - Installing symfony/var-dumper (v3.4.22): Downloading (100%)
  - Installing nikic/php-parser (v4.2.0): Downloading (100%)
  - Installing jakub-onderka/php-console-color (v0.2): Downloading (100%)
  - Installing jakub-onderka/php-console-highlighter (v0.4): Downloading (100%)
  - Installing dnoegel/php-xdg-base-dir (0.1): Downloading (100%)
  - Installing psy/psysh (v0.9.9): Downloading (100%)
  - Installing webmozart/assert (1.4.0): Downloading (100%)
  - Installing webmozart/path-util (2.3.0): Downloading (100%)
  - Installing webflo/drupal-finder (1.1.0): Downloading (100%)
  - Installing symfony/filesystem (v3.4.22): Downloading (100%)
  - Installing symfony/config (v3.4.22): Downloading (100%)
  - Installing stecman/symfony-console-completion (0.9.0): Downloading (100%)
  - Installing drupal/console-en (1.8.0): Downloading (100%)
  - Installing dflydev/placeholder-resolver (v1.0.2): Downloading (100%)
  - Installing dflydev/dot-access-data (v1.1.0): Downloading (100%)
  - Installing dflydev/dot-access-configuration (v1.0.3): Downloading (100%)
  - Installing drupal/console-core (1.8.0): Downloading (100%)
  - Installing alchemy/zippy (0.4.3): Downloading (100%)
  - Installing drupal/console (1.8.0): Downloading (100%)
  - Installing symfony/phpunit-bridge (v3.4.22): Downloading (100%)
  - Installing sebastian/recursion-context (3.0.0): Downloading (100%)
  - Installing doctrine/instantiator (1.1.0): Downloading (100%)
  - Installing sebastian/exporter (3.1.0): Downloading (100%)
  - Installing sebastian/diff (2.0.1): Downloading (100%)
  - Installing sebastian/comparator (2.1.3): Downloading (100%)
  - Installing phpdocumentor/reflection-common (1.0.1): Downloading (100%)
  - Installing phpdocumentor/type-resolver (0.4.0): Downloading (100%)
  - Installing phpdocumentor/reflection-docblock (4.3.0): Downloading (100%)
  - Installing phpspec/prophecy (1.8.0): Downloading (100%)
  - Installing sebastian/version (2.0.1): Downloading (100%)
  - Installing sebastian/resource-operations (1.0.0): Downloading (100%)
  - Installing sebastian/object-reflector (1.1.1): Downloading (100%)
  - Installing sebastian/object-enumerator (3.0.3): Downloading (100%)
  - Installing sebastian/global-state (2.0.0): Downloading (100%)
  - Installing sebastian/environment (3.1.0): Downloading (100%)
  - Installing phpunit/php-text-template (1.2.1): Downloading (100%)
  - Installing phpunit/phpunit-mock-objects (5.0.10): Downloading (100%)
  - Installing phpunit/php-timer (1.0.9): Downloading (100%)
  - Installing phpunit/php-file-iterator (1.4.5): Downloading (100%)
  - Installing theseer/tokenizer (1.1.0): Downloading (100%)
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Downloading (100%)
  - Installing phpunit/php-token-stream (2.0.2): Downloading (100%)
  - Installing phpunit/php-code-coverage (5.3.2): Downloading (100%)
  - Installing phar-io/version (1.0.1): Downloading (100%)
  - Installing phar-io/manifest (1.0.1): Downloading (100%)
  - Installing myclabs/deep-copy (1.8.1): Downloading (100%)
  - Installing phpunit/phpunit (6.5.14): Downloading (100%)
  - Installing mikey179/vfsstream (v1.6.5): Downloading (100%)
  - Installing jcalderonzumba/gastonjs (v1.2.0): Downloading (100%)
  - Installing behat/mink (dev-master d5ee350): Cloning d5ee350c40 from cache
    Failed to download behat/mink from source: Failed to clone https://github.com/minkphp/Mink.git via https, ssh protocols, aborting.

- https://github.com/minkphp/Mink.git
  Cloning into '/home/ag/public_html/composertest/site/vendor/behat/mink'...
  done.
  fatal: unable to create thread: Resource temporarily unavailable
  fatal: cannot repack to clean up

- git@github.com:minkphp/Mink.git
  Cloning into '/home/ag/public_html/composertest/site/vendor/behat/mink'...
  done.
  fatal: unable to create thread: Resource temporarily unavailable
  fatal: cannot repack to clean up

    Now trying to download from dist
  - Installing behat/mink (dev-master d5ee350): Downloading (100%)
  - Installing jcalderonzumba/mink-phantomjs-driver (v0.3.3): Downloading (100%)
  - Installing squizlabs/php_codesniffer (3.4.0): Downloading (100%)
  - Installing drupal/coder (8.3.1): Cloning 29a25627e7 from cache


  [RuntimeException]
  Failed to execute git clone --no-checkout '/home/ag/.cache/composer/vcs/https---git.drupal.org-project-coder.git/' '/home/ag/public_html/composertest/site/vendor/drupal/coder'
   --dissociate --reference '/home/agplume/.cache/composer/vcs/https---git.drupal.org-project-coder.git/' && cd '/home/ag/public_html/composertest/site/vendor/drupal/coder' && git re
  mote set-url origin 'https://git.drupal.org/project/coder.git' && git remote add composer 'https://git.drupal.org/project/coder.git'


create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-reqs] [--] [<package>] [<directory>] [<version>]

```

**Comment by Asterovim** [comment]
I have tested with composer 1.7.2 and Composer 1.8.4. 
composer clearcache don't change nothing.

**Comment by Asterovim** [comment]
I do not know why what has changed (i do nothing..) but it works today.

**Comment by VioletFlare** [comment]
I had a similar error today with composer v1.0. In my case there was a problem with windows credentials since my domain credentials were updated. I tried to change my Credentials through Windows Credentials Manager, but to no avail (maybe a relog was actually required to refresh the cache or something, this is what might have happened to the OP). In the end I just deleted the entry for git credentials in the Windows Credentials Manager and then I cloned the repository that was failing to download. I just ran 

`git clone <httpsrepourl>`

and re-entered my credentials.

After that I managed to run

`composer install`

just fine.
